test/views/manageBusinesses/add/IncomeSourceAddedObligationsViewSpec.scala

```diff
@@ -62,128 +79,884 @@ class IncomeSourceAddedObligationsViewSpec extends ViewSpec {
+            case Some(upcomingDeadlineMessage) =>
+              upcomingDeadlineMessage.text() shouldBe "Your next quarterly update for the 2024 to 2025 tax year is due by 5 August 2024 for the quarterly period 6 April 2024 to 5 July 2024"
+              upcomingDeadlineMessage.select("b").text() shouldBe "5 August 2024"
+            case _ => fail("No upcoming quarterly obligation text was found.")
+          }
+
+          Option(document.getElementById("obligations-list")) match {
+            case Some(upcomingDeadlineMessage) =>
+              upcomingDeadlineMessage.text() shouldBe "Your tax return for the 2024 to 2025 tax year is due by 31 January 2026"
+              upcomingDeadlineMessage.select("b").text() shouldBe "31 January 2026"
+            case _ => fail("No upcoming annual obligation text was found.")
+          }
+        }
+      }
     }
 
-    "Display previous tax year message" in new Setup(validCallWithData) {
-      val prevYearsSection: Element = layoutContent.getElementById("prevYears")
-      prevYearsSection.text() should include(IncomeSourceAddedMessages.prevYearsHeading)
-      prevYearsSection.text() should include(IncomeSourceAddedMessages.prevYearsText)
+    "Display the view upcoming updates link when there are no overdue obligations" in new Setup(validCallWithData) {
+      Option(document.getElementById("view-upcoming-updates")) match {
+        case Some(upcomingUpdatesLink) => upcomingUpdatesLink.text() shouldBe "View your upcoming updates"
+        case None => fail("No upcoming updates link was found")
+      }
+    }
+    "Display the view overdue and upcoming updates link when there are no overdue obligations" in new Setup(validCurrentTaxYearQuarterlyCallOneOverdue) {
+      Option(document.getElementById("view-upcoming-updates")) match {
+        case Some(upcomingUpdatesLink) => upcomingUpdatesLink.text() shouldBe "View your overdue and upcoming updates"
+        case None => fail("No upcoming updates link was found")
+      }
     }
 
-    "Not display any obligation sections when user has no obligations" in new Setup(validUKPropertyBusinessCall) {
-      Option(layoutContent.getElementById("quarterly")) shouldBe None
-      Option(layoutContent.getElementById("prevyears")) shouldBe None
+    "Render the view all your business link" in new Setup(validCallWithData) {
+      document.getElementById("view-all-businesses-link").text() shouldBe IncomeSourceAddedMessages.viewAllBusinessesText
+      document.getElementById("view-all-businesses-link").select("a").attr("href") shouldBe manageBusinessesUrl
     }
 
-    "render the your income sources button" in new Setup(validCallWithData) {
-      document.getElementById("continue-button").text() shouldBe (IncomeSourceAddedMessages.buttonText)
+    "Display the correct change reporting frequency text" when {
+      // TODO the links of these tests will need to change to the new entry point for the opt in/out journeys once the page is made
```

Comment:
TODO is here to change the link when the new opt in/out entry point is developed.